src/client.js

```diff
@@ -36,38 +36,41 @@ function initSocket() {
 
 global.socket = initSocket();
 
-const component = (
-  <Router render={(props) =>
-        <ReduxAsyncConnect {...props} helpers={{client}} filter={item => !item.deferred} />
-      } history={history}>
-    {getRoutes(store)}
-  </Router>
-);
-
-ReactDOM.render(
-  <Provider store={store} key="provider">
-    {component}
-  </Provider>,
-  dest
-);
-
-if (process.env.NODE_ENV !== 'production') {
-  window.React = React; // enable debugger
-
-  if (!dest || !dest.firstChild || !dest.firstChild.attributes || !dest.firstChild.attributes['data-react-checksum']) {
-    console.error('Server-side React render was discarded. Make sure that your initial render does not contain any client-side code.');
-  }
-}
+const routes = getRoutes(store);
+match({ history, routes: routes }, () => {
```

Comment:
Good question. `match` is going to wait until React Router has finished getting components. In other words, React Router will fetch chunks and  tell us via `match` when it's ready to render the entire view.
If we don't wait then React Router will render the view without chunks and that view will be different than what we got from server side rendering. This will cause React to discard SSR content.